common/variant/src/main/java/org/apache/spark/types/variant/Variant.java

```diff
@@ -316,6 +334,22 @@ static void toJsonImpl(byte[] value, byte[] metadata, int pos, StringBuilder sb,
       case BINARY:
         appendQuoted(sb, Base64.getEncoder().encodeToString(VariantUtil.getBinary(value, pos)));
         break;
+      case YEAR_MONTH_INTERVAL:
+        IntervalFields ymFields = VariantUtil.getYearMonthIntervalFields(value, pos);
+        int ymValue = (int) VariantUtil.getLong(value, pos);
```

Comment:
shouldn't we call `getInt`?

Reply:
`getLong` in `VariantUtils.java` looks at the first byte in the variant and decides which type of int should be extracted. Therefore, functions like `getInt` or `getShort` don't exist. I have added cases in `getLong` for `YEAR_MONTH_INTERVAL` and `DAY_TIME_INTERVAL`.